geopyspark/geotrellis/rasterize.py

```diff
@@ -29,13 +30,33 @@ def rasterize(geoms, crs, zoom, fill_value, cell_type=CellType.FLOAT64, options=
         crs = str(crs)
 
     pysc = get_spark_context()
-    wkb_geoms = [shapely.wkb.dumps(g) for g in geoms]
-    srdd = pysc._gateway.jvm.geopyspark.geotrellis.SpatialTiledRasterLayer.rasterizeGeometry(pysc._jsc.sc(),
-                                                                                             wkb_geoms,
-                                                                                             crs,
-                                                                                             zoom,
-                                                                                             float(fill_value),
-                                                                                             CellType(cell_type).value,
-                                                                                             options,
-                                                                                             num_partitions)
+    rasterizer = pysc._gateway.jvm.geopyspark.geotrellis.SpatialTiledRasterLayer.rasterizeGeometry
+
+    if isinstance(geoms, (list, tuple)):
+        wkb_geoms = [dumps(g) for g in geoms]
+
+        srdd = rasterizer(pysc._jsc.sc(),
+                          wkb_geoms,
+                          crs,
+                          zoom,
+                          float(fill_value),
+                          CellType(cell_type).value,
+                          options,
+                          num_partitions)
+
+    else:
+        wkb_rdd = geoms.map(lambda geom: dumps(geom))
+
+        # If this is False then the WKBs will be serialized
+        # when going to Scala resulting in garbage
+        wkb_rdd._bypass_serializer = True
```

Comment:
Nice find. Looks like this bypasses the normal use of python serializers and allows the `JavaRDD` to get at the WKB bytes directly?

Reply:
Thanks. Yeah, or else the WKBs will be serialized with what ever serializer that `RDD` has which is unhelpful in this case.